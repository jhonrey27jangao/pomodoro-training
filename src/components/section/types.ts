export type SectionProps = {
  children?: any;
};
