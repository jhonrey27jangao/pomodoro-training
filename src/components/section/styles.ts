import styled from "styled-components";

export const StyledSection = styled.section`
  box-shadow: 0 1rem 4rem 0;
  color: #ffffff;
  position: relative;
  text-align: center;
`;

export const StyledImageDiv = styled.div`
  min-height: 300px;
  top: 0;
  left: 0;
  overflow: hidden;
  z-index: -1;
  padding: 20px;

  &:after {
    -moz-transition: opacity 1.3125s ease-in-out;
    -webkit-transition: opacity 1.3125s ease-in-out;
    -ms-transition: opacity 1.3125s ease-in-out;
    transition: opacity 1.3125s ease-in-out;
    content: "";
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-image: url(images/overlay.png), url(images/overlay.png),
      linear-gradient(45deg, #7b64b5, #b66e66);
    opacity: 0;
    z-index: 1;
  }

  &:before {
    -moz-transition: opacity 1.3125s ease-in-out;
    -webkit-transition: opacity 1.3125s ease-in-out;
    -ms-transition: opacity 1.3125s ease-in-out;
    transition: opacity 1.3125s ease-in-out;
    content: "";
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-image: url(images/overlay.png), url(images/overlay.png),
      linear-gradient(45deg, #a480ff, #ff8f83);
    opacity: 0.7;
    z-index: 1;
  }
`;

export const StyledDivContent = styled.div`
  position: relative;
  z-index: 2;
  margin: 0 auto;
  color: #7a7d86;
`;
