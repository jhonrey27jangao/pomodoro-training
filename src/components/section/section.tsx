import React from "react";
import { StyledSection, StyledImageDiv, StyledDivContent } from "./styles";
import { SectionProps } from "./types";
export const Section: React.FC<SectionProps> = ({ children }) => (
  <StyledSection>
    <StyledImageDiv>
      <StyledDivContent>{children}</StyledDivContent>
    </StyledImageDiv>
  </StyledSection>
);
