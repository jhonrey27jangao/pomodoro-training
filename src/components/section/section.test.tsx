import React from "react";
import { render } from "@testing-library/react";
import { Section } from "./section";

describe("Test for Section Component", () => {
  it("should render Section", () => {
    const { container } = render(<Section />);
    expect(container).toMatchSnapshot();
  });
});
