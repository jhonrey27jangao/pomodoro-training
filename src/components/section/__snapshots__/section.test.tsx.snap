// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Section Component should render Section 1`] = `
<div>
  <section
    class="sc-AxjAm LyrjF"
  >
    <div
      class="sc-AxirZ cRKUMI"
    >
      <div
        class="sc-AxiKw eopyYK"
      />
    </div>
  </section>
</div>
`;
